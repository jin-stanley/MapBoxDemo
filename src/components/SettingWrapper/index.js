/***
 * Using for FilterButton and LayersButton
 */
import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { ReactComponent as CloeseLogo } from "../../img/svg/x.svg";

const SettingWrapper = (props) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      <div className={classes.header}>
        <p>{props.name}</p>
        <CloeseLogo
          className={classes.closeIcon}
          onClick={() => props.handleCloseSetting()}
        />
      </div>
      <div className={classes.content}>{props.children}</div>
    </div>
  );
};

const useStyles = makeStyles({
  container: {
    minWidth: "320px",
    // maxHeight: "420px",
    position: "absolute",
    bottom: 124,
    right: 16,
    zIndex: 3,
    borderRadius: "8px",
    background: "#FFFFFF",
  },
  header: {
    fontWeight: 800,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    borderTopLeftRadius: "8px",
    borderTopRightRadius: "8px",
    background: "#1F4B5C",
    color: "#fff",
    padding: "4px 0",
    position: "relative",
  },
  content: {
    height: "100%",
    padding: "8px",
  },
  closeIcon: {
    position: "absolute",
    right: 16,
    width: "12px",
    height: "12px",
    fill: "#fff",
    cursor: "pointer",
  },
});

export default SettingWrapper;
