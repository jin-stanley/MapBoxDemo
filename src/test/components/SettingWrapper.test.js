import React from "react";
import renderer from "react-test-renderer";

import SettingWrapper from "../../components/SettingWrapper";

it("SettingWrapper can render success", () => {
  const tree = renderer.create(<SettingWrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
