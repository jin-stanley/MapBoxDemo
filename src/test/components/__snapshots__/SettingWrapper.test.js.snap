// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingWrapper can render success 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-header-2"
  >
    <p />
    <svg
      className="makeStyles-closeIcon-4"
      onClick={[Function]}
    >
      x.svg
    </svg>
  </div>
  <div
    className="makeStyles-content-3"
  />
</div>
`;
